packages/app/src/runner/event-manager.ts

```diff
@@ -451,6 +453,12 @@ export class EventManager {
             Cypress.runner.resumeAtTest(runState.currentId, runState.emissions)
           }
 
+          // this sends the beforeDriverRun hook to protocol in the case where
+          // top has reloaded due to a mid-spec origin change
+          if (Cypress.config('protocolEnabled')) {
+            return this.ws.emit('protocol:before:driver:run', run)
+          }
```

Comment:
Suggest moving this into the above if statement where we check for a `currentId`:

```suggestion
            Cypress.runner.resumeAtTest(runState.currentId, runState.emissions)
            
            // this sends the beforeDriverRun hook to protocol in the case where
            // top has reloaded due to a mid-spec origin change
            if (Cypress.config('protocolEnabled')) {
              return this.ws.emit('protocol:before:driver:run', run)
            }
          }
```

Reply:
This needs to always run if this function doesn't return early above, so that protocol gets `beforeDriverRun` reliably before the driver runs.